core/backend/src/ExportGraphics.ts

```diff
@@ -464,12 +464,93 @@ export class ExportGraphicsMeshVisitor extends PolyfaceData implements PolyfaceV
+    return ExportGraphicsMeshSubsetVisitor.createSubsetVisitor(this._polyface, facetIndices, numWrap);
+  }
+}
 
+/**
+ * An `ExportGraphicsMeshSubsetVisitor` is an `ExportGraphicsMeshVisitor` which only visits a subset of the facets.
+ * * The subset is defined by an array of facet indices provided when this visitor is created.
+ * * Input indices (e.g., for `moveToReadIndex`) are understood to be indices into the subset array.
+ * @public
+ */
+export class ExportGraphicsMeshSubsetVisitor extends ExportGraphicsMeshVisitor {
+  private _facetIndices: number[];
+  private _currentSubsetIndex: number;    // index within _parentFacetIndices, or -1 after construction
+  private _nextSubsetIndex: number;       // index within _parentFacetIndices
+
+  private constructor(polyface: ExportGraphicsMesh, facetIndices: number[], numWrap: number) {
+    super(polyface, numWrap);
+    this._facetIndices = facetIndices.slice();
+    this._currentSubsetIndex = -1;
+    this._nextSubsetIndex = 0;
+  }
+  private isValidSubsetIndex(index: number): boolean {
+    return index >= 0 && index < this._facetIndices.length;
+  }
+  /**
+   * Create a visitor for iterating a subset of the facets of `polyface`.
+   * @param polyface reference to the client polyface, supplying facets
+   * @param facetIndices array of indices of facets in the client polyface to visit. This array is cloned.
+   * @param numWrap number of vertices replicated in the visitor arrays to facilitate simpler caller code. Default is zero.
+   */
+  public static createSubsetVisitor(
+    polyface: ExportGraphicsMesh, facetIndices: number[], numWrap: number = 0,
+  ): ExportGraphicsMeshSubsetVisitor {
+    return new ExportGraphicsMeshSubsetVisitor(polyface, facetIndices, numWrap);
+  }
+  /**
+   * Advance the iterator to a particular facet in the subset of client polyface facets.
+   * @param subsetIndex index into the subset array, not to be confused with the client facet index.
+   * @return whether the iterator was successfully moved.
+   */
+  public override moveToReadIndex(subsetIndex: number): boolean {
+    if (this.isValidSubsetIndex(subsetIndex)) {
+      this._currentSubsetIndex = subsetIndex;
+      this._nextSubsetIndex = subsetIndex + 1;
+      return super.moveToReadIndex(this._facetIndices[subsetIndex]);
+    }
+    return false;
+  }
+  /**
+   * Advance the iterator to the next facet in the subset of client polyface facets.
+   * @return whether the iterator was successfully moved.
+   */
+  public override moveToNextFacet(): boolean {
+    if (this._nextSubsetIndex !== this._currentSubsetIndex)
+      return this.moveToReadIndex(this._nextSubsetIndex);
+    this._nextSubsetIndex++;
+    return true;
+  }
+  /** Call this before iterating facets with [[moveToNextFacet]]. */
+  public override reset(): void {
+    this.moveToReadIndex(0);
+    this._nextSubsetIndex = 0; // so immediate moveToNextFacet stays here.
```

Comment:
For one line in code comments I dont put `.` at the end
```suggestion
    this._nextSubsetIndex = 0; // so immediate moveToNextFacet stays here
```